### PR TITLE
Add CI to trigger a deployment to NPM

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Merging Giacomo's CI changes to `main` branch

This change will allow us to publish to npm from github UI

### CODE REVIEW

File: src/workflows/publish.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+fail_if_unset GITLAB_TRIGGER_TOKEN
+fail_if_unset BUCKET_NAME
+fail_if_unset PUBLISH_SERVER
+fail_if_unset AWS_ACCESS_KEY_ID
+fail_if_unset AWS_SECRET_ACCESS_KEY
+fail_if_unset AWS_REGION
+
+git_username="GitHub Actions"
+git_useremail="GitHub-Actions@GPv2-contracts"
```
Comment (fedgiac) [inline src/workflows/publish.sh]: :stuck_out_tongue: 

```suggestion
git_useremail="GitHub-Actions@cow-sdk"
```

File: src/workflows/publish.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  echo "Tag $version_tag is already present"
+  exit 1
+fi
+
+yarn pack --filename package.tgz
+
+aws configure set aws_access_key_id "$AWS_ACCESS_KEY_ID"
+aws configure set aws_secret_access_key "$AWS_SECRET_ACCESS_KEY"
+aws configure set region "$AWS_REGION"
+if ! aws s3 cp package.tgz "s3://$BUCKET_NAME/cow-token/cow-token-$version.tgz"; then
```
Comment (fedgiac) [inline src/workflows/publish.sh]:
```suggestion
if ! aws s3 cp package.tgz "s3://$BUCKET_NAME/cow-sdk/cow-sdk-$version.tgz"; then
```

**Review by fedgiac — COMMENTED** [review]
Just two small issues, otherwise it looks good!

**Review by fedgiac — APPROVED** [review]

**Comment by anxolin** [comment]
Cool!

**Pull request merged**